modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/SpringCodegen.java

```diff
@@ -579,6 +588,251 @@ public void preprocessOpenAPI(OpenAPI openAPI) {
+                  String opId = getOrGenerateOperationId(op.getValue(), e.getKey(), op.getKey().toString());
+                  // process request body
+                  RequestBody b = ModelUtils.getReferencedRequestBody(openAPI, op.getValue().getRequestBody());
+                  Schema requestSchema = null;
+                  if (b != null) {
+                    requestSchema = ModelUtils.getSchemaFromRequestBody(b);
+                  }
+                  if (requestSchema != null) {
+                    schemas.put(opId, requestSchema);
+                  }
+                  if (op.getValue().getResponses() != null){
+                    // process all response bodies
+                    for (Map.Entry<String, ApiResponse> ar : op.getValue().getResponses().entrySet()) {
+                      ApiResponse a = ModelUtils.getReferencedApiResponse(openAPI, ar.getValue());
+                      Schema responseSchema = ModelUtils.getSchemaFromResponse(a);
+                      if (responseSchema != null) {
+                        schemas.put(opId + ar.getKey(), responseSchema);
+                      }
+                    }
+                  }
+                }
+              }
+            }
+
+        for (Map.Entry<String, Schema> e : schemas.entrySet()) {
+          String n = toModelName(e.getKey());
+          Schema schema = e.getValue();
+          String nOneOf = toModelName(n + "OneOf");
+          if (ModelUtils.isComposedSchema(schema)) {
+            List<String> names = new ArrayList<>();
+            ComposedSchema composedSchema = (ComposedSchema) schema;
+            if (composedSchema.getOneOf() != null && composedSchema.getOneOf().size() > 0){
+              for (Schema oneOfSchema : composedSchema.getOneOf()) {
+                names.add(getSingleSchemaType(oneOfSchema));
+              }
+              String name = "OneOf" + String.join("", names);
+              addOneOfNameExtension(schema, name);
+              addOneOfInterfaceModel((ComposedSchema) schema, name);
+            }
+          } else if (ModelUtils.isArraySchema(schema)) {
+                Schema items = ((ArraySchema) schema).getItems();
+                if (ModelUtils.isComposedSchema(items)) {
+                  addOneOfNameExtension(items, nOneOf);
+                  addOneOfInterfaceModel((ComposedSchema) items, nOneOf);
+                }
+          } else if (ModelUtils.isMapSchema(schema)) {
+                    Schema addProps = ModelUtils.getAdditionalProperties(schema);
+                    if (addProps != null && ModelUtils.isComposedSchema(addProps)) {
+                      addOneOfNameExtension(addProps, nOneOf);
+                      addOneOfInterfaceModel((ComposedSchema) addProps, nOneOf);
+                    }
+                }
+            }
+        }
+    }
+
+    public void addOneOfNameExtension(Schema s, String name) {
```

Comment:
AFAICS, this is the same as `DefaultCodegen.addOneOfNameExtension` and so I think it can be safely omitted.